### ISSUE TITLE
Build toolchain for DPDK development

### ISSUE THREAD

**Issue by daniel-noland** [githedgehog/fabric#608]
Create a container which builds DPDK and all relevant dependencies

- [ ] with llvm or gcc compilers (ideally the exact same compiler which built our kernel for whatever test we want to run)
- and in debug mode, produces code
  - [x] with debug symbols (`-ggdb3` and or `-glldb`) 
  - [x] `-Og` compiler optimizations
  - [x] with source code of dpdk and all dependencies packaged in that container for static analysis
- in release mode (gnu toolchain)
  - [ ] produces detached debug symbols
  - [x] `-O3` optimizations
  - [x] `-flto` (likely with `-ffat-lto-objects`
  - [x] links with `ld` or `mold`
- in release mode (llvm toolchain)
  - [ ] produces detached debug symbols
  - [x]  `-O3` optimizations
  - [x] `-flto=thin` (likely with `-ffat-lto-objects -funified-lto`) 
  - [x] links with `mold` (rdma-core is being a troll)

The main idea with the different toolchains is to

1. aid debug (sometimes one compiler's error messages are _far_ more helpful than anothers)
2. avoid dependencies which are either 
  - unmaintained
  - code to implementation rather than spec
  - overly opinionated or inflexible
 
Basically, if you refuse to compile on `clang` or you can only link with a specific linker or use GNU specific C extensions then I am suspicious of your library :)

**Comment by daniel-noland** [comment]
I think we should move the detached debug into a different task

**Comment by daniel-noland** [comment]
Moving distinct toolchains to a different issue as it isn't currently pressing
